src/CacheTower/CacheStack.cs

```diff
@@ -378,5 +378,28 @@ public async ValueTask DisposeAsync()
 
 			Disposed = true;
 		}
+
+		/// <summary>
+		/// Synchronously disposes the current instance of <see cref="CacheStack"/> and all associated layers and extensions.
+		/// </summary>
+		public void Dispose()
+		{
+			if (Disposed)
+			{
+				return;
+			}
+
+			foreach (var layer in CacheLayers)
+			{
+				if (layer is IDisposable disposableLayer)
```

Comment:
We might actually have an issue here - if someone calls the synchronous dispose method but an extension doesn't expose a synchronous dispose method itself, it may not be properly disposed.